Add test outlier detection

chiara-arpae opened the pull request:
[ERAD]Test that input arrays with more than 2 dimensions raise a ValueError exception](https://app.codecov.io/gh/pySTEPS/pysteps/compare/283/tree/pysteps/utils/cleansing.py#D1L173)

chiara-arpae committed:
Add test outlier detection

katelbach commented on pysteps/tests/test_utils_cleansing.py:
```diff
@@ -174,3 +174,9 @@ def test_detect_outlier_multivariate_local():
+    V = np.zeros((20, 3, 2))
+    thr_std_devs = 1
+    with pytest.raises(ValueError):
+        cleansing.detect_outliers(V, thr_std_devs, k=10)
```
did you by chance forget the coord parameter? according to documentation it should not be set to None if k is passed

aperezhortal replied:
Hi @katelbach, another option is to remove the `k=10` argument. It is not needed for the test.

katelbach commented on pysteps/tests/test_utils_cleansing.py:
```diff
@@ -174,3 +174,9 @@ def test_detect_outlier_multivariate_local():
     assert outliers.sum() == 2
+
+def test_detect_outlier_input_dims():
+    V = np.zeros((20, 3, 2))
```
could you test a list of input dimensions/different cases?

katelbach reviewed (changes requested):
thanks a lot for your contribution, i proposed a few modifications to your code

aperezhortal reviewed (changes requested):
Thanks @chiara-arpae for the contribution. 

The test is failing because there is a bug in the `detect_outliers` function. 
When the dimensions are incorrect, the code raising the exception has a error:
```
raise ValueError(
    f"input_array must have 1 (n) or 2 dimensions (n, m), "
    f"but it has {coord.ndim}"
)
```
Here, `{coord.ndim}` should be `{input_array.ndim}` 

Do you want to add the fix to bug?

dnerini commented:
hey @chiara-arpae please let us know if something is not clear in the comments left above. The pull request is almost ready to be merged, just few final changes are needed ;)

chiara-arpae commented:
Hi! I'm very sorry for late. I should have committed the suggested correction few minutes ago. Please let me know if it's all fine.

chiara-arpae committed:
Update cleansing.py

correcting input_array instead of coords in log message

chiara-arpae commented:
Hi, I'm very sorry for late. I should have committed the correction but I
received a message about check black failing from Github.
I realized I have modified and committed on master branch, instead of doing
it on the branch 'test_cleaning_2d'. Maybe could it depend on that? Sorry
very much for mess. If you could help me I'd be grateful.


Il giorno ven 16 set 2022 alle ore 10:32 Daniele Nerini <
***@***.***> ha scritto:

> hey @chiara-arpae <https://github.com/chiara-arpae> please let us know if
> something is not clear in the comments left above. The pull request is
> almost ready to be merged, just few final changes are needed ;)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pySTEPS/pysteps/pull/298#issuecomment-1249080374>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQ2PKGSGINQDM24BD7VMYZDV6QWCFANCNFSM573BL2AA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

chiara-arpae committed:
Update test_utils_cleansing.py

removed k parameter from cleansing.detect_outliers in test.test_utils_cleansing.test_detect_outlier_input_dims() because not necessary for test and coords is not passed.

dnerini commented:
hey @chiara-arpae no worries and thanks for getting back on this! It looks good to go! :-)
